src/rtp_transceiver/mod.rs

```diff
@@ -348,6 +348,11 @@ impl RTCRtpTransceiver {
                 previous,
                 d
             );
+
+            let sender = self.sender.lock().await;
+            if let Some(sender) = &*sender {
```

Comment:
The `direction` of a Transceiver reflects the desired state, as such we shouldn't change anything about sending/receiving here. Instead it should be based on the `current_direction` which reflects the actually negotiated value. 

The appropriate place to set the sender as paused or not is in `process_new_current_direction`.

Reply:
There's a TODO for this on line 422

Reply:
ah alright. wasn't aware of that one.